cmd/syft/cli/commands/attest.go

```diff
@@ -0,0 +1,258 @@
+	"github.com/anchore/syft/cmd/syft/cli/options"
+	"github.com/anchore/syft/internal"
+	"github.com/anchore/syft/internal/bus"
+	"github.com/anchore/syft/internal/file"
+	"github.com/anchore/syft/internal/log"
+	"github.com/anchore/syft/syft/event"
+	"github.com/anchore/syft/syft/event/monitor"
+	"github.com/anchore/syft/syft/formats"
+	"github.com/anchore/syft/syft/formats/github"
+	"github.com/anchore/syft/syft/formats/syftjson"
+	"github.com/anchore/syft/syft/formats/table"
+	"github.com/anchore/syft/syft/formats/template"
+	"github.com/anchore/syft/syft/formats/text"
+	"github.com/anchore/syft/syft/sbom"
+	"github.com/anchore/syft/syft/source"
+)
+
+const (
+	attestExample = `  {{.appName}} {{.command}} --output [FORMAT] alpine:latest defaults to using images from a Docker daemon. If Docker is not present, the image is pulled directly from the registry
+`
+	attestSchemeHelp = "\n  " + schemeHelpHeader + "\n" + imageSchemeHelp
+	attestHelp       = attestExample + attestSchemeHelp
+)
+
+type attestOptions struct {
+	options.SingleOutput `yaml:",inline" mapstructure:",squash"`
+	options.UpdateCheck  `yaml:",inline" mapstructure:",squash"`
+	options.Packages     `yaml:",inline" mapstructure:",squash"`
+	options.Attest       `yaml:",inline" mapstructure:",squash"`
+}
+
+func Attest(app clio.Application) *cobra.Command {
+	var allowableOutputs []string
+	for _, f := range formats.AllIDs() {
+		switch f {
+		case table.ID, text.ID, github.ID, template.ID:
+			continue
+		}
+		allowableOutputs = append(allowableOutputs, f.String())
+	}
+
+	opts := &attestOptions{
+		UpdateCheck: options.UpdateCheckDefault(),
+		SingleOutput: options.SingleOutput{
+			AllowableOptions: allowableOutputs,
+			Output:           syftjson.ID.String(),
+		},
+		Packages: options.PackagesDefault(),
+	}
+
+	return app.SetupCommand(&cobra.Command{
+		Use:   "attest --output [FORMAT] <IMAGE>",
+		Short: "Generate an SBOM as an attestation for the given [SOURCE] container image",
+		Long:  "Generate a packaged-based Software Bill Of Materials (SBOM) from a container image as the predicate of an in-toto attestation that will be uploaded to the image registry",
+		Example: internal.Tprintf(attestHelp, map[string]interface{}{
+			"appName": app.ID().Name,
+			"command": "attest",
+		}),
+		Args: validatePackagesArgs,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			applicationUpdateCheck(app, opts.CheckForAppUpdate)
+
+			return runAttest(app, opts, args[0])
+		},
+	}, opts)
+}
+
+//nolint:funlen
+func runAttest(app clio.Application, opts *attestOptions, userInput string) error {
+	_, err := exec.LookPath("cosign")
+	if err != nil {
+		// when cosign is not installed the error will be rendered like so:
+		// 2023/06/30 08:31:52 error during command execution: 'syft attest' requires cosign to be installed: exec: "cosign": executable file not found in $PATH
+		return fmt.Errorf("'syft attest' requires cosign to be installed: %w", err)
+	}
+
+	defer bus.Exit()
```

Comment:
all of these `bus.Exit()` lines are subtle and have implications with initialization logic (so must always be run). I think it would be easier to think about this if they were moved to the wrapper function [here](https://github.com/anchore/syft/blob/abd1b43547bb4ff90cde9fcf880f57f47f4532c1/cmd/syft/cli/commands/attest.go#L74-L77). 

For instance, where it is now, it is possible to not have cosign installed and not call exit.

This also has the benefit of separating bus implementation concerns with business concerns.